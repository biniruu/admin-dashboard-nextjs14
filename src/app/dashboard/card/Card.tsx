import { MdSupervisedUserCircle } from 'react-icons/md'

import styles from './card.module.css'

function Card() {
  return (
    <div className={styles.container}>
      <MdSupervisedUserCircle size={24} />
      <div className={styles.texts}>
        <span className={styles.title}>total users</span>
        <span className={styles.number}>10.273</span>
        <span className={styles.detail}>
          <span className={styles.positive}>12%</span>
          more than
        </span>
      </div>
    </div>
  )
}

export default Card
