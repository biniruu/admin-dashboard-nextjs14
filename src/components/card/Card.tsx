import { type CardData } from 'types'

function Card({ icon, title, number, state, rate, detail }: CardData) {
  return (
    <div className="flex w-full cursor-pointer gap-5 rounded bg-bg-soft p-5 hover:bg-hover">
      {icon}
      <div className="flex flex-col gap-5">
        <div>{title}</div>
        <div className="text-2xl font-medium">{number}</div>
        <div className="text-sm font-light">
          <span className={`${state === 'positive' ? 'text-lime' : 'text-red'}`}>{rate}%</span>
          {detail}
        </div>
      </div>
    </div>
  )
}

export default Card
