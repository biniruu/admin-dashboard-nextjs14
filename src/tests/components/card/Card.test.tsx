import { render, screen } from '@testing-library/react'
import { MdSupervisedUserCircle } from 'react-icons/md'
import renderer from 'react-test-renderer'

import Card from '@components/card/Card'
import type { CardData } from 'types'

jest.mock('react-icons/md', () => ({ MdSupervisedUserCircle: () => <div>iconğŸš€</div> }))

describe('Card component', () => {
  const data: CardData = {
    icon: <MdSupervisedUserCircle />,
    title: 'title',
    number: 1,
    state: 'positive',
    rate: 1,
    detail: 'detail',
  }

  test('should render Card component correctly', () => {
    const result = renderer.create(<Card data={data} />).toJSON()

    expect(result).toMatchSnapshot()
  })

  test('should have "text-lime" class when the state prop is "positive"', () => {
    render(<Card data={data} />)

    expect(screen.getByTestId('rate').classList.contains('text-lime')).toBe(true)
  })
})
