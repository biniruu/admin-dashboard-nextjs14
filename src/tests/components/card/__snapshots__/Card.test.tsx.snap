// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render Card component correctly 1`] = `
<div
  className="flex w-full cursor-pointer gap-5 rounded bg-bg-soft p-5 hover:bg-hover"
>
  <div>
    iconðŸš€
  </div>
  <div
    className="flex flex-col gap-5"
  >
    <div>
      title
    </div>
    <div
      className="text-2xl font-medium"
    >
      1
    </div>
    <div
      className="text-sm font-light"
    >
      <span
        className="text-lime pr-2"
        data-testid="rate"
      >
        1
        %
      </span>
      detail
    </div>
  </div>
</div>
`;
